web/src/main/java/org/hillview/InitialObjectTarget.java

```diff
@@ -136,10 +136,14 @@ void prepareFiles(RpcRequest request, RpcRequestContext context) {
         } else if (which == 4) {
             fileNamePattern = "segmentation.csv";
             schemaFile = "segmentation.schema";
+        } else if (which == 5) {
+            config.separator = '\t';
+            fileNamePattern = "criteoTab.gz";
+            schemaFile = "criteo.schema";
         } else {
             throw new RuntimeException("Unexpected operation " + which);
-        }
-
+		}
```

Comment:
Please don't use tabs. You should configure your editor to always use spaces.
If it is emacs I can send you some lisp code.